src/main/scala/it/pagopa/interop/attributeregistryprocess/service/impl/AttributeRegistryManagementServiceImpl.scala

```diff
@@ -0,0 +1,101 @@
+package it.pagopa.interop.attributeregistryprocess.service.impl
+
+import com.typesafe.scalalogging.{Logger, LoggerTakingImplicit}
+import it.pagopa.interop.attributeregistrymanagement.client.invoker.{ApiError, ApiRequest, BearerToken}
+import it.pagopa.interop.attributeregistrymanagement.client.model.{Attribute, AttributeSeed, AttributesResponse}
+import it.pagopa.interop.commons.logging.{CanLogContextFields, ContextFieldsToLog}
+import it.pagopa.interop.commons.utils.withHeaders
+import it.pagopa.interop.attributeregistryprocess.error.AttributeRegistryProcessErrors.RegistryAttributeNotFound
+import it.pagopa.interop.attributeregistryprocess.service.{
+  AttributeRegistryManagementApi,
+  AttributeRegistryManagementInvoker,
+  AttributeRegistryManagementService
+}
+
+import scala.concurrent.{ExecutionContext, Future}
+import java.util.UUID
+
+final case class AttributeRegistryManagementServiceImpl(
+  invoker: AttributeRegistryManagementInvoker,
+  api: AttributeRegistryManagementApi
+)(implicit ec: ExecutionContext)
+    extends AttributeRegistryManagementService {
+
+  implicit val logger: LoggerTakingImplicit[ContextFieldsToLog] =
+    Logger.takingImplicit[ContextFieldsToLog](this.getClass)
+
+  override def getAttributeById(id: UUID)(implicit contexts: Seq[(String, String)]): Future[Attribute] = withHeaders {
+    (bearerToken, correlationId, ip) =>
+      val request: ApiRequest[Attribute] =
+        api.getAttributeById(xCorrelationId = correlationId, attributeId = id, xForwardedFor = ip)(
+          BearerToken(bearerToken)
+        )
+      invoker.invoke(request, s"Retrieving Attribute $id").recoverWith {
+        case err: ApiError[_] if err.code == 404 =>
+          Future.failed(RegistryAttributeNotFound(id.toString))
+      }
+  }
+
+  override def createAttribute(
+    attributeSeed: AttributeSeed
+  )(implicit contexts: Seq[(String, String)]): Future[Attribute] = withHeaders { (bearerToken, correlationId, ip) =>
+    val request: ApiRequest[Attribute] =
+      api.createAttribute(xCorrelationId = correlationId, attributeSeed = attributeSeed, xForwardedFor = ip)(
+        BearerToken(bearerToken)
+      )
+    invoker.invoke(request, s"Attribute with name ${attributeSeed.name} created")
```

Comment:
ℹ️ this log is not referred to the successfully completed operation, but it's just a generic description of the operation.
Better change it to avoid misleading logs
```suggestion
    invoker.invoke(request, s"Attribute creation with name ${attributeSeed.name}")
```